### ISSUE TITLE
Splash-as-a-proxy tests are broken in Python 2.7.10

### ISSUE THREAD

**Issue by kmike** [scrapinghub/splash#241]
Tests are failing:

```
_____________________________________________________ ProxyRunJsTest.test_js_and_console ______________________________________________________
[gw3] darwin -- Python 2.7.10 /Users/kmike/envs/splash/bin/python2.7
self = <splash.tests.test_proxy_server.ProxyRunJsTest testMethod=test_js_and_console>

    def test_js_and_console(self):
        js_source = """function test(x){ return x; }
    console.log('some log');
    console.log('another log');
    test('abc');"""
        params = {'console': '1'}
>       r = self._runjs_request(js_source, params=params).json()

splash/tests/test_runjs.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
splash/tests/test_proxy_server.py:142: in _runjs_request
    return self.request(query, endpoint=endpoint)
splash/tests/test_render.py:92: in request
    return self._get_handler().request(query, endpoint, headers, **kwargs)
splash/tests/test_proxy_server.py:26: in request
    return requests.get(url, headers=headers, proxies=proxies)
/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/requests/api.py:68: in get
    return request('get', url, **kwargs)
/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/requests/api.py:50: in request
    response = session.request(method=method, url=url, **kwargs)
/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/requests/sessions.py:464: in request
    resp = self.send(prep, **send_kwargs)
/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/requests/sessions.py:576: in send
    r = adapter.send(request, **kwargs)
/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/requests/adapters.py:370: in send
    timeout=timeout
/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:544: in urlopen
    body=body, headers=headers)
/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py:349: in _make_request
    conn.request(method, url, **httplib_request_kw)
/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py:1053: in request
    self._send_request(method, url, body, headers)
/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py:1092: in _send_request
    self.putheader(hdr, value)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <requests.packages.urllib3.connection.HTTPConnection object at 0x10ee49090>
header = 'x-splash-js-source'
values = ["function test(x){ return x; }\nconsole.log('some log');\nconsole.log('another log');\ntest('abc');"]
v = "function test(x){ return x; }\nconsole.log('some log');\nconsole.log('another log');\ntest('abc');"
one_value = "function test(x){ return x; }\nconsole.log('some log');\nconsole.log('another log');\ntest('abc');"

    def putheader(self, header, *values):
        """Send a request header line to the server.
    
            For example: h.putheader('Accept', 'text/html')
            """
        if self.__state != _CS_REQ_STARTED:
            raise CannotSendHeader()
    
        header = '%s' % header
        if not _is_legal_header_name(header):
            raise ValueError('Invalid header name %r' % (header,))
    
        values = [str(v) for v in values]
        for one_value in values:
            if _is_illegal_header_value(one_value):
>               raise ValueError('Invalid header value %r' % (one_value,))
E               ValueError: Invalid header value "function test(x){ return x; }\nconsole.log('some log');\nconsole.log('another log');\ntest('abc');"

/usr/local/Cellar/python/2.7.10/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py:1031: ValueError
```

**Comment by kmike** [comment]
It is caused by a fix to this bug: http://bugs.python.org/issue22928. It is not safe to use header values with new lines (besides some specific cases), so Python stdlib disallows it now. The problem is that  `X-Splash-js-source` header often contains such values, so maybe it was not a good idea to introduce it in first place.

**Comment by kmike** [comment]
Closed by #367.
